File: test/net/dinkla/raytracer/utilities/TimerTest.java
package net.dinkla.raytracer.utilities;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;


/**
 * Created by IntelliJ IDEA.
 * User: Jörn Dinkla
 * Date: 14.04.2010
 * Time: 14:23:01
 * To change this template use File | Settings | File Templates.
 */
public class TimerTest {

    @Test
    public void get() throws Exception {
        Timer t = new Timer();

        t.start();
        Thread.sleep(5);
        t.stop();
        final long x = t.getDuration();
        assertTrue(x > 0);

        t.start();
        Thread.sleep(50);
        t.stop();
        final long y = t.getDuration();
        assertTrue(y > 0);

        assertTrue(y > x);

    }

}

File: src/test/kotlin/net/dinkla/raytracer/utilities/TimerTest.kt
package net.dinkla.raytracer.utilities

import net.dinkla.raytracer.interfaces.Timer
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class TimerTest {

    @Test
    @Throws(Exception::class)
    fun get() {
        val t = Timer()

        t.start()
        Thread.sleep(5)
        t.stop()
        val x = t.duration
        assertTrue(x > 0)

        t.start()
        Thread.sleep(50)
        t.stop()
        val y = t.duration
        assertTrue(y > 0)

        assertTrue(y > x)
    }

}
